packages/core/src/Editor.ts

```diff
@@ -112,14 +111,8 @@ export class Editor extends EventEmitter<EditorEvents> {
     this.on('focus', this.options.onFocus)
     this.on('blur', this.options.onBlur)
     this.on('destroy', this.options.onDestroy)
-
-    if (this.options.onPaste) {
-      this.registerPlugin(PastePlugin(this.options.onPaste))
-    }
-
-    if (this.options.onDrop) {
-      this.registerPlugin(DropPlugin(this.options.onDrop))
-    }
+    this.on('drop', ({ event, slice, moved }) => this.options.onDrop(event, slice, moved))
+    this.on('paste', ({ event, slice }) => this.options.onPaste(event, slice))
```

Comment:
I figured, that while I was here, I may as well promote these to be actual editor events that follow the same pattern as others. This allows flexibility to register multiple handlers to listen to the same event

Reply:
@nperez0111 This resolves [this](https://github.com/ueberdosis/tiptap/discussions/5613), right?

Reply:
@svenadlung it does 😉